File: CS/DiagnosticTool/Data/BenchmarkHelper.cs
using System.Collections.Generic;
using System.IO;
using System.Xml.Serialization;

namespace DashboardDiagnosticTool.Data {
    class BenchmarkHelper {
        public static void SaveBenchmarkItems(Dictionary<int, List<BenchmarkItem>> benchmarks, Dictionary<int, Dictionary<int, List<TraceItem>>> traceItems, IEnumerable<SessionItem> sessions, Stream stream) {
            var serializer = new XmlSerializer(typeof(BenchmarkItemSet));
            serializer.Serialize(stream, new BenchmarkItemSet(benchmarks, traceItems, sessions));
        }
        public static BenchmarkItemSet LoadBenchmarkSet(Stream stream) {
            var serializer = new XmlSerializer(typeof(BenchmarkItemSet), null, null, null, null);
            return serializer.Deserialize(stream) as BenchmarkItemSet;
        }
    }
}

File: VB/DiagnosticTool/Data/BenchmarkHelper.vb
Imports System.Collections.Generic
Imports System.IO
Imports System.Xml.Serialization

Namespace DashboardDiagnosticTool.Data

    Friend Class BenchmarkHelper

        Public Shared Sub SaveBenchmarkItems(ByVal benchmarks As Dictionary(Of Integer, List(Of BenchmarkItem)), ByVal traceItems As Dictionary(Of Integer, Dictionary(Of Integer, List(Of TraceItem))), ByVal sessions As IEnumerable(Of SessionItem), ByVal stream As Stream)
            Dim serializer = New XmlSerializer(GetType(BenchmarkItemSet))
            serializer.Serialize(stream, New BenchmarkItemSet(benchmarks, traceItems, sessions))
        End Sub

        Public Shared Function LoadBenchmarkSet(ByVal stream As Stream) As BenchmarkItemSet
            Dim serializer = New XmlSerializer(GetType(BenchmarkItemSet), Nothing, Nothing, Nothing, Nothing)
            Return TryCast(serializer.Deserialize(stream), BenchmarkItemSet)
        End Function
    End Class
End Namespace
